### PR TITLE
[quant] Add missing functions and methods for channelwise quantization

### PR DESCRIPTION
**Pull request by dskhudia**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24936 [quant] per channel quantization support
* #24935 [quant] Per Channel quantization APIs
* **#24934 [quant] Add missing functions and methods for channelwise quantization**

1) Functions and methods to get scales and zero_points for channelwise quantization were missing. Adding these.
2) Correctly print quantized tensors for channelwise quantization.

Differential Revision: [D16659715](https://our.internmc.facebook.com/intern/diff/D16659715/)

### CODE REVIEW

File: aten/src/ATen/core/Tensor.h [hunk 1]
```diff
@@ -640,6 +640,8 @@ class CAFFE2_API Tensor {
   Tensor dequantize() const;
   double q_scale() const;
   int64_t q_zero_point() const;
+  Tensor q_scales() const;
```
Comment (raghuramank100) [inline aten/src/ATen/core/Tensor.h]: Lets call it something more distinctive: q_per_axis_scale and q_per_axis_zero_point? The current names differ by one letter and are unclear

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -1898,6 +1898,22 @@ def callable(a, b) -> number
...
+               r"""
+q_scales() -> Tensor
+
+Given a Tensor quantized by linear (affine) per-channel quantization,
+returns a Tensor of scales of the underlying quantizer().
+""")
+
+add_docstr_all('q_zero_points',
+               r"""
+q_zero_points() -> Tensor
```
Comment (raghuramank100) [inline torch/_tensor_docs.py]: Please change the names to be consistent (see comment on q_scales and q_zero_points)

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Some comments

File: aten/src/ATen/core/Tensor.h (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/core/Tensor.h]: should be per_channel I think

**Comment by dskhudia** [comment]
> Some comments

Addressed comments.

File: test/test_quantized_tensor.py [hunk 3]
```diff
@@ -52,13 +52,15 @@ def test_qtensor(self):
         qr[:] = x
         self.assertEqual(qr.item(), 15)
         # we can also print a qtensor
-        self.assertEqual(str(qr),
+        self.assertEqual(' '.join(str(qr).split()),
```
Comment (raghuramank100) [inline test/test_quantized_tensor.py]: Please add a test for q_per_channel_scales() and q_per_channel_zero_points()

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Please add tests for per channel quant, thanks

File: test/test_quantized_tensor.py (hunk 3 above)
Comment (dskhudia) [reply test/test_quantized_tensor.py]: Test for q_per_channel_scales() and q_per_channel_zero_points() are added in the followup diff. Please checkout 'test_per_channel_qtensor_creation' here: https://github.com/pytorch/pytorch/pull/24935/files 

quantize_per_channel and dequanize() already exists. Please check test_qtensor_per_channel_affine in this same file (test/test_quantized_tensor.py).

**Comment by dskhudia** [comment]
@raghuramank100 : The tests exist. Please see my comment.

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**